File: 0002-add-two-numbers/0002-add-two-numbers.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        
        ListNode l3 = new ListNode();
        ListNode head = l3;
        
        int carry = 0;
       
        while (l1 != null || l2 != null){
            int firstNo = 0, secondNo = 0;
            
            
            if (l1 != null){
                firstNo = l1.val;
            }
            if (l2!=null){
                secondNo = l2.val;
            }
            
            l3.val = (firstNo + secondNo + carry)%10;
            carry = (firstNo + secondNo + carry)/10;
            
            
            if (l1 != null && l1.next != null){
                l1 = l1.next;
             
            }else{
                l1 = null;
            }
            
            if (l2!= null && l2.next != null){
                l2 = l2.next;
            }else{
                l2 = null;
            }
            
            
            if (l1 != null || l2 != null){
                
            
            l3.next = new ListNode();
            l3 = l3.next;
            }else{
                if (carry >0) {
                l3.next = new ListNode();
            l3 = l3.next;
                l3.val += carry;
            }}
          
            
        }
        
        return head;
    }
}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        
    
        r = 0
        head = ListNode(0, None)
        rhead = head
        last = None
        while l1 and l2:
            
            res = l1.val + l2.val + r
            head.val = res % 10
            r = res // 10
            head.next = ListNode(0, None)

            last = head
            head = head.next
            l1 = l1.next
            l2 = l2.next

        
        while l1:
            res = l1.val + r
            head.val = res % 10
            r = res // 10

            head.next = ListNode(0, None)

            last = head
            l1 = l1.next
            head = head.next

        while l2:
            res = l2.val + r
            head.val = res % 10
            r = res // 10

            last = head
            head.next = ListNode(0, None)
            head = head.next
            l2 = l2.next

        if r != 0:
            head.val = r
        else:
            last.next = None
        return rhead